packages/ui-components/src/components/search.tsx

```diff
@@ -215,6 +224,35 @@ export const FilterBox = (props: IFilterBoxProps): JSX.Element => {
 /**
  * A widget which hosts a input textbox to filter on file names.
  */
-export const FilenameSearcher = (props: IFilterBoxProps): ReactWidget => {
-  return ReactWidget.create(<FilterBox {...props} />);
+class FilenameSearcherWidget extends ReactWidget {
+  constructor(props: IFilterBoxProps) {
+    super();
+    this._filterBoxProps = { ...props };
+    props?.filterSettingsChanged?.connect((_, args) => {
+      this._updateProps(args);
+    }, this);
+  }
+
+  render(): JSX.Element {
+    return <FilterBox {...this._filterBoxProps} />;
+  }
+
+  /**
+   * Update a specific prop.
+   */
+  private _updateProps(props: Partial<IFilterBoxProps>): void {
+    Object.assign(this._filterBoxProps, props);
+    this.update();
+  }
+
+  private _filterBoxProps: IFilterBoxProps;
+}
+
+/**
+ * Function which returns a widget that hosts an input textbox to filter on file names.
+ */
+export const FilenameSearcher = (
+  props: IFilterBoxProps
+): FilenameSearcherWidget => {
```

Comment:
While annotating it as returning `FilenameSearcherWidget` is better, especially in new code, here I would be conservative and keep it as it was if possible (which I think is possible) just not to break downstream users when backporting this fix.

```suggestion
export const FilenameSearcher = (props: IFilterBoxProps): ReactWidget => {
```